devicetypes/stelpro/stelpro-ki-zigbee-thermostat.src/stelpro-ki-zigbee-thermostat.groovy

```diff
@@ -0,0 +1,687 @@
+            }
+		}
+        else if (descMap.cluster == "0201" && descMap.attrId == "401c") {
+            log.debug "SETPOINT MODE"
+            log.debug "descMap.value $descMap.value"
+            map.name = "thermostatMode"
+            map.value = modeMap[descMap.value]
+            map.data = [supportedThermostatModes: supportedThermostatModes]
+            //sendEvent(name:"thermostatMode", value:map.value, data:[supportedThermostatModes: supportedThermostatModes])
+		}
+        else if (descMap.cluster == "0201" && descMap.attrId == "0008") {
+        	log.debug "HEAT DEMAND"
+            map.name = "thermostatOperatingState"
+            if (descMap.value < "10") {
+            	map.value = "idle"
+            }
+            else {
+            	map.value = "heating"
+            }
+            //sendEvent(name:"thermostatOperatingState", value:map.value)
+            if (settings.heatdetails == "No") {
+    			map.displayed = false
+    		}
+        }
+	}
+
+	def result = null
+	if (map) {
+		result = createEvent(map)
+	}
+	log.debug "Parse returned $map"
+	return result
+}
+
+def parseDescriptionAsMap(description) {
+	(description - "read attr - ").split(",").inject([:]) { map, param ->
+		def nameAndValue = param.split(":")
+		map += [(nameAndValue[0].trim()):nameAndValue[1].trim()]
+	}
+}
+
+def updateWeather() {
+	log.debug "updating weather"
+	def weather
+    // If there is a zipcode defined, weather forecast will be sent. Otherwise, no weather forecast.
+	if (settings.zipcode) {
+        log.debug "ZipCode: ${settings.zipcode}"
+        weather = getWeatherFeature( "conditions", settings.zipcode )
+
+        // Check if the variable is populated, otherwise return.
+        if (!weather) {
+            log.debug( "Something went wrong, no data found." )
+            return false
+        }
+        
+        def tempToSend
+		
+        if(getTemperatureScale() == "C" ) {
+        	tempToSend = weather.current_observation.temp_c
+            log.debug( "Outdoor Temperature: ${tempToSend} C" )
+        }
+        else {
+        	tempToSend = weather.current_observation.temp_f
+            log.debug( "Outdoor Temperature: ${tempToSend} F" )
+        }       
+        sendEvent( name: 'outsideTemp', value: tempToSend )
+        quickSetOutTemp(tempToSend)
+    }
+}
+
+def getModeMap() { [
+	"00":"off",
+    "04":"heat",
+	"05":"eco"
+]}
+
+def getFanModeMap() { [
+	"04":"fanOn",
+	"05":"fanAuto"
+]}
+
+def poll() {
+    return pollCalled()
+}
+
+def pollCalled() {
+    delayBetween([
+    			updateWeather(),
+                zigbee.readAttribute(0x201, 0x0000),	//Read Local Temperature
+                zigbee.readAttribute(0x201, 0x0008),	//Read PI Heating State
+    			zigbee.readAttribute(0x201, 0x0012),	//Read Heat Setpoint
+                zigbee.readAttribute(0x201, 0x001C),	//Read System Mode
+                my_readAttribute(0x201, 0x401C, ["mfgCode": "0x1185"]),	//Read Manufacturer Specific Setpoint Mode
```

Comment:
Manufacturer specific reads can be done with this command:

`zigbee.readAttribute(0x0201, 0x401C, [mfgCode: 0x1185])`

Reply:
Is this new?
Manufacturer specific had to be read separately before.